### PR TITLE
Set up pixi dependency management with testing and linting environments

### LINKED ISSUE #34: Set up pixi dependency management with testing and linting environments
No, I wouldn't install dependencies via pre-commit.

Furthermore, I suggest to create testing, linting and other dependency environments so that you can install them with scoped versions from those files instead of adding them in the CI like here.

We can have a look at that in our call :)

_Originally posted by @Zaubeerer in https://github.com/Tanner-Ray-Martin/Dynamic-DB/pull/32#discussion_r1714025063_

- [x] review issue requirements with @Zaubeerer on call

- [ ] create testing, linting, and other dependency environments

### PR DESCRIPTION
**Pull request by Tanner-Ray-Martin**
Fixes #34

added pixi environments (default), test, and format

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -166,3 +166,7 @@ cython_debug/
 # pixi environments
 .pixi
 *.egg-info
+
+
+#Databases
```
Comment (Tanner-Ray-Martin) [inline .gitignore:171]: added sqlite3 to gitignore

File: pyproject.toml [hunk 2]
```diff
@@ -18,9 +18,9 @@ platforms = ["win-64"]
 Dynamic-DB = { path = ".", editable = true }
 
 [tool.pixi.tasks]
-poc = "uvicorn POC.api.main:app --reload"
-test_pre_commit = "pre-commit run --all-files"
-test_poc_mypy = "mypy --check-untyped-defs --strict"
+poc = "uvicorn POC.api.main:app"
```
Comment (Tanner-Ray-Martin) [inline pyproject.toml:21]: removed reload for default environment and added reload to test environment.

File: pyproject.toml [hunk 3]
```diff
@@ -30,6 +30,21 @@ sqlmodel = ">=0.0.21,<0.1"
...
 
+[tool.pixi.feature.test.tasks]
+test = "pytest"
+pre_commit = "pre-commit run --all-files"
+mypy_poc = "mypy POC --check-untyped-defs --ignore-missing-imports"
+
+[tool.pixi.feature.format.tasks]
+format = "ruff"
+check = "ruff check"
+fix = "ruff check --fix"
```
Comment (Tanner-Ray-Martin) [inline pyproject.toml:41]: added various ruff commands to format environment

File: pyproject.toml (hunk 2 above)
Comment (Zaubeerer) [reply pyproject.toml:21]: Nice, great that you know how to leverage the different commands 👍

**Review by Zaubeerer — APPROVED** [review]
Looks good to me :)

Do you have any questions regarding features and environments?

**Comment by Tanner-Ray-Martin** [comment]
no questions at this time! :) 💯

**Pull request merged**